### ISSUE TITLE
In mobile, selecting selectOneMenu item with ajax update of inputText hangs with spinning cursor

### ISSUE THREAD

**Issue by steinarb** [primefaces/primefaces#1864]
# IMPORTANT
- If you open a issue, fill every item. Otherwise the issue might be closed as invalid.
- Before you open a issue, test it with the current/newest version.
## 1) Environment
- PrimeFaces version: 5.3, 6.0, 6.1-SNAPSHOT
- Does it work on the newest PrimeFaces version?
  No. I first encountered the issue in version 5.3, tried upgrading to 6.0, encountered the problem there as well, and tried 6.1-SNAPSHOT checked out from github and built locally and it didn't work there either.
- Application server + version:
  karaf 4.0.7 with pax-web 4.3.0
- Affected browsers:
  Vivaldi 1.4.589.38, android chrome, IE 11 (I suspect all browsers I'll try)
## 2) Expected behavior

When selecting an item in the selectOneMenu I expected the inputText to be updated with a numeric value
## 3) Actual behavior

When selecting an item in the selectOneMenu, the inputText loses its styling, and the cursor is spinning endlessly.

I have verified in the debugger that the bean property setter of the selectOneMenu is called, and that the bean property getter for the inputText is called.
## 4) Steps to reproduce

1, get the test code
```
git clone https://github.com/steinarb/primefaces.mobile.update.inputtext.git
cd primefaces.mobile.update.inputtext
mvn install
```

2. Download and unpack karaf from http://www.apache.org/dyn/closer.lua/karaf/4.0.7/apache-karaf-4.0.7.zip

3. start karaf
```
   cd apache-karaf-4.0.7
   bin\karaf.bat debug
```

4. Give the following commands to the karaf prompt to install and run the test application
```
   feature:repo-add mvn:no.priv.bang.testcases/primefaces.mobile.update.inputtext/LATEST/xml/features
   feature:install pax-jsf-support
   feature:install updatetest
```

5. Visit http://localhost:8181/testapp/ and select an item from "Choose job"

## 5) Sample XHTML

https://github.com/steinarb/primefaces.mobile.update.inputtext/blob/master/src/main/webapp/index.xhtml

``` xml
<?xml version="1.0" encoding="ISO-8859-1" ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html lang="en" xmlns="http://www.w3.org/1999/xhtml"
      xmlns:f="http://java.sun.com/jsf/core"
      xmlns:h="http://java.sun.com/jsf/html"
      xmlns:ui="http://java.sun.com/jsf/facelets"
      xmlns:p="http://primefaces.org/ui"
      xmlns:pm="http://primefaces.org/mobile">

 <f:view renderKitId="PRIMEFACES_MOBILE" />

 <h:head>
 </h:head>

 <h:body>
  <pm:page id="page">
   <pm:header title="Select job"></pm:header>

   <pm:content>

    <h:form id="test">
     <p:panelGrid columns="3" styleClass="ui-responsive" >
      <p:selectOneMenu id="jobtype" value="#{testbean.newJobType}" converter="jobConverter">
       <f:selectItem itemLabel="Choose job" itemValue="" noSelectionOption="true" />
       <f:selectItems value="#{testbean.jobTypes}" var="jobtype2" itemValue="#{jobtype2}" itemLabel="#{jobtype2.name}"/>
       <p:ajax event="change" update="amount" listener="#{testbean.newJobTypeSelected}"/>
      </p:selectOneMenu>
      <h:inputText id="amount" value="#{testbean.newJobWages}" required="true" readonly="true"/>
      <p:commandButton value="Registrer jobb" update="jobtype amount" actionListener="#{testbean.registerNewJob}" />
     </p:panelGrid>
    </h:form>
   </pm:content>
  </pm:page>
 </h:body>
</html>
```

## 6) Sample bean

https://github.com/steinarb/primefaces.mobile.update.inputtext/blob/master/src/main/java/selecttest/

``` java
@ManagedBean(name = "testbean")
@ViewScoped
public class TestBean {

    private List<Job> jobTypes;
    Job newJobType;
    double newJobWages;

    public TestBean() {
        jobTypes = Arrays.asList(
                                 new Job("Vacuum", 35.0),
                                 new Job("Take out trash", 25.0),
                                 new Job("Do the dishes", 75.0)
                                 );
    }

    public Job getNewJobType() {
        return newJobType;
    }

    public void setNewJobType(Job newJobType) {
        this.newJobType = newJobType;
        if (newJobType != null) {
            setNewJobWages(newJobType.getAmount());
        }
    }

    public double getNewJobWages() {
        return newJobWages;
    }

    public void setNewJobWages(double newJobWages) {
        this.newJobWages = newJobWages;
    }

    public List<Job> getJobTypes() {
        return jobTypes;
    }

    public void newJobTypeSelected(final AjaxBehaviorEvent event) {
        if (newJobType != null) {
            newJobWages = newJobType.getAmount();
        }
    }

    public void registerNewJob(ActionEvent event) {
        clearNewJobValues();
    }

    private void clearNewJobValues() {
        setNewJobType(null);
        setNewJobWages(0);
    }
}
```

**Comment by tandraschko** [comment]
Sry but why do we need karaf? Normally users provide examples with maven/jetty ;)

**Comment by steinarb** [comment]
I spent half Sunday trying to get an eclipse/tomcat setup working so that I could test the test application, and finally gave up and created the application in a form that works for me. :-)

But the important parts are the xhtml page and the bean, and they are self-contained, so changing the POM by removing the maven-bundle-plugin config should create a regular war.

I could do that but I don't have a setup that would let me verify that the end result works.

Here's the bug repro: https://github.com/steinarb/primefaces.mobile.update.inputtext

(Note that karaf uses jetty, so the behaviour will be jetty-like, and note also that this bug probably is in the JavaScript layer so the web/appserver used shouldn't make any difference.)

**Comment by steinarb** [comment]
Here's a primefaces mobile forum thread where I discuss this problem with myself: http://forum.primefaces.org/viewtopic.php?f=8&t=47803

Note the similarities to Issue #632 (but that issue should already be fixed in 6.0 or 6.1-SNAPSHOT...?).

**Comment by steinarb** [comment]
Note also that debugging the beans from eclipse is extremely easy if karaf is started in the way described in the example (ie. with the "debug" argument):
1. Import the testcase as a maven project in eclipse https://github.com/primefaces/primefaces/issues/1864#issuecomment-257247874
2. Create a remote debug configuration in eclipse that connects to port 5005 localhost and looks in the imported testcase maven project for the source code
3. Run the remote debug configuration and set breakpoints in the bean methods

**Comment by steinarb** [comment]
The following javascript stack trace appears in the chrome console when selecting an item in the selectOneMenu:

```
jquery.js.xhtml?ln=primefaces&v=6.0&v=6.0:25 Uncaught DOMException: Failed to execute 'replaceChild' on 'Node': The node to be replaced is not a child of this node.(…)
(anonymous function) @ jquery.js.xhtml?ln=primefaces&v=6.0&v=6.0:25
            domManip @ jquery.js.xhtml?ln=primefaces&v=6.0&v=6.0:25
         replaceWith @ jquery.js.xhtml?ln=primefaces&v=6.0&v=6.0:25
       updateElement @ core.js.xhtml?ln=primefaces&v=6.0&v=6.0:3
            doUpdate @ core.js.xhtml?ln=primefaces&v=6.0&v=6.0:3
              handle @ core.js.xhtml?ln=primefaces&v=6.0&v=6.0:3
             success @ core.js.xhtml?ln=primefaces&v=6.0&v=6.0:3
                   i @ jquery.js.xhtml?ln=primefaces&v=6.0&v=6.0:25
            fireWith @ jquery.js.xhtml?ln=primefaces&v=6.0&v=6.0:25
                  cf @ jquery.js.xhtml?ln=primefaces&v=6.0&v=6.0:25
                   i @ jquery.js.xhtml?ln=primefaces&v=6.0&v=6.0:25
```

**Comment by steinarb** [comment]
I have tried debugging this in chrome.

In the method below when I'm ready to step into the replaceChild() function and click on the "i" and "this" variables, they highlight the same <input> element in the document structure.  Clicking on the e variable highlights the <div> element that is the parent of the <input> in the document tree.

Is this an attempt to replace an object with itself?

I can't step into the replaceChild() call with F11. When I press F11 the javascript fails there and the replaceChild call gets a red zig-zag underlining.

``` javascript
        replaceWith: function() {
            var e = arguments[0];
            this.domManip(arguments, function(i) {
                e = this.parentNode;
                bH.cleanData(l(this));
                if (e) {
                    e.replaceChild(i, this)
                }
            });
```

**Comment by tandraschko** [comment]
PrimeFaces Mobile will be deprecated  in 6.2 and removed in 6.3 ( see https://www.primefaces.org/primefaces-6-2-roadmap/)
Therefor i will close this issue as won't fix. If you will provide a PR to fix this issue until 6.2 RC1, i can merge it.
